### ISSUE TITLE
Problem "invalid group" with v5.24.4 and later

### ISSUE THREAD

**Issue by IonVillarreal** [casbin/node-casbin#441]
Hi I updated the library to the latest version and I got this error which seems to be caused by #439 in version 5.24.4

Attached the capture of the error in NextJS

<img width="745" alt="Captura de pantalla 2023-03-23 a la(s) 17 11 23" src="https://user-images.githubusercontent.com/12217695/227362674-0767cf9f-317f-41af-882f-0c5d019e504b.png">

Someone else happens ? I'll stay on version 5.24.3

**Comment by hsluoyz** [comment]
@IonVillarreal plz provide the full code, Casbin model, policy, request and response (expected and got)

**Comment by IonVillarreal** [comment]
Hi, sorry for the delay, I attached the snippet of the code where the error happened

```
import Head from "next/head";
import Image from "next/image";
import { Inter } from "next/font/google";
import styles from "@/styles/Home.module.css";
import { useEffect } from "react";
import {
  Enforcer,
  newEnforcer,
  newModelFromString,
  StringAdapter,
} from "casbin";


export const basicModel = `
[request_definition]
r = sub, obj, act

[policy_definition]
p = sub, obj, act

[policy_effect]
e = some(where (p.eft == allow))

[matchers]
m = (r.sub == p.sub || p.sub == "*") && keyMatch2(r.obj, p.obj) && regexMatch(r.act, p.act)
`;
export const basicPolicy = `g, admin, ADMINISTRADOR`;

export default function Home() {
  const iniciar = async () => {
    const politicas: string[][] = []; // TODO: add
    const model = newModelFromString(basicModel);
    const policy = new StringAdapter(basicPolicy);
    const enforcerTemp: Enforcer = await newEnforcer(model, policy);
    for await (const p of politicas) {
      await enforcerTemp.addPolicy(p[0], p[1], p[2], p[3], p[4], p[5]);
    }
    return enforcerTemp;
  };

  useEffect(() => {
    iniciar();
  }, []);

  return (
    <>
      <Head>
        <title>Create Next App</title>
      </Head>
      <main className={styles.main}>
        <div className={styles.description}>
          <div>
            <a
              href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
              target="_blank"
              rel="noopener noreferrer"
            >
              By{" "}
              <Image
                src="/vercel.svg"
                alt="Vercel Logo"
                className={styles.vercelLogo}
                width={100}
                height={24}
                priority
              />
            </a>
          </div>
        </div>
      </main>
    </>
  );
}
```

![image](https://user-images.githubusercontent.com/12217695/228079543-6e2d3156-8ae9-40c6-91d2-6373df7f86b1.png)

The error happens in Safari v13.3, but not happen in Safari v13.4 that released today

**Comment by IonVillarreal** [comment]
Stills happen in v5.26.0 in Safari v16.3 but not still happen in Safari 16.4

**Comment by hsluoyz** [comment]
@johnsonjo4531 please take a look as it is caused by: https://github.com/casbin/node-casbin/pull/439

**Comment by johnsonjo4531** [comment]
Yes, this is definitely my fault 😅. I used an [ES2018 feature RegExp Negative LookBehind](https://github.com/tc39/proposal-regexp-lookbehind#negative-assertions) to implement the two lines of RegExp I used in my PR. I should have let you all know that it wouldn't work on older browsers. It should work on all modern (non End Of Life) Node.js versions though as far as I'm aware. I don't have much time to assist with the fix. But [these two lines in ./src/util/util.ts are what need to be changed to not use back references.](https://github.com/casbin/node-casbin/pull/439/commits/3fda526ef5138314d5ec15663f3cfa970d08f6e3#diff-7ebe731ef2444c83f2cfe4a7f1bbd722e1dcb2392904dc8983c95a5741b787caR21) If they are changed back to what they used to be they will fail the tests I added to fix the bug in #438. See this series of answers on stack overflow for possible answers to not use the [negative lookbehind](https://stackoverflow.com/a/11347100/2066736).
